src/main/java/org/apache/maven/shared/incremental/IncrementalBuildHelper.java

```diff
@@ -61,17 +69,28 @@
     private MavenProject mavenProject;
 
     /**
-     * Used for detecting changes between the Mojo execution.
+     * Used for detecting changes in the output directory between the Mojo execution.
      * @see #getDirectoryScanner();
      */
-    private DirectoryScanner directoryScanner;
+    private DirectoryScanner directoryScanner = new DirectoryScanner();
```

Comment:
This class is deprecated. Consider using use java.nio.file.DirectoryStream or java.nio.Files.walkFileTree() and related classes instead. Those should definitely be safer in a multi-threaded environment.